contract/pkg/client.go

```diff
@@ -67,6 +85,112 @@ func CreateBlockchainClient(apiUrl string) BlockchainClient {
 	}
 }
 
+func (b *blockchainClient) SetEventDispatcher(contractAddressSS58 string, dispatcher map[types.Hash]ContractEventDispatchEntry) error {
+	contract, err := DecodeAccountIDFromSS58(contractAddressSS58)
+	if err != nil {
+		return err
+	}
+	b.eventContractAccount = contract
+	b.eventDispatcher = dispatcher
+	err = b.listenContractEvents()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (b *blockchainClient) listenContractEvents() error {
+	meta, err := b.RPC.State.GetMetadataLatest()
+	if err != nil {
+		return err
+	}
+
+	key, err := types.CreateStorageKey(meta, "System", "Events", nil, nil)
+	if err != nil {
+		return err
+	}
+
+	sub, err := b.RPC.State.SubscribeStorageRaw([]types.StorageKey{key})
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := signal.NotifyContext(context.Background(), syscall.SIGHUP, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
+	b.eventContextCancel = cancel
+	watchdog := time.NewTicker(time.Minute)
+	eventArrived := true
+	go func() {
+		defer sub.Unsubscribe()
+		defer log.Info("Exited") // TODO: remove before merge
+		for {
+			select {
+			case <-ctx.Done():
+				log.Info("Exitting...") // TODO: remove before merge
+				return
+
+			case <-watchdog.C:
+				if !eventArrived {
+					s, err := b.RPC.State.SubscribeStorageRaw([]types.StorageKey{key})
```

Comment:
Good point. Looks, in case of short outage, we may iterate block that we missed and handle its events.
I'll put it into the backlog, looks this scenario is rare.